src/compressor/zstd/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+# zstd
+
+# libzstd - build it statically
+set(ZSTD_EXTRA_CFLAGS -fPIC -Wno-unused-variable -O3)
+
+add_custom_target(build_zstd
+  COMMAND $(MAKE) libzstd CFLAGS="${ZSTD_EXTRA_CFLAGS}"
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src/zstd/lib
+  COMMENT "libzstd building")
+
+add_library(zstd STATIC IMPORTED)
+set_property(TARGET zstd PROPERTY IMPORTED_LOCATION "${CMAKE_SOURCE_DIR}/src/zstd/lib/libzstd.a")
```

Comment:
@cbodley @alimaredia would someone with more cmake mojo than I have mind take a stab at it?